src/taxonomy/api/hooks/selectors.ts

```diff
@@ -0,0 +1,14 @@
+import * as api from './api';
+import { TaxonomyListData } from '../types';
+
+export const useTaxonomyListDataResponse = (): TaxonomyListData | undefined => {
```

Comment:
@bradenmacdonald It's missing to configure `jest` to run tests on Typescript files. I ran out of time here. can we do that configuration and the missing test on a folow-up ticket?

Reply:
**Note:** The functionality in the typescript files is not complex, it is just the connection to the API

Reply:
@ChrisChV In that case, for now you could make this a `.js` file and [put all the types as JSDoc comments](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#providing-type-hints-in-js-via-jsdoc). Then TypeScript can still be run to check the types in exactly the same way, IDEs like VS Code will show all the types, but you don't have to change `jest` or `webpack` or anything else.

But yes it would be nicer to get `jest` and `.ts` files working properly in a future story if possible. We also have to hear from the maintainers of this repo to make sure they're on board with it too. But I'm sure JSDoc are welcome.

Reply:
@bradenmacdonald Yes, also I feel like the typescript webpack/jest configuration is now brittle and can break anytime later. So, yes, I like the JSDoc approach. I updated the files to use that aproach and I enabled the check only on the taxonomy api files; and I added that check on the `make validate` workflow